www/js/sections.js

```diff
@@ -165,6 +177,9 @@ const applyTransformationsToFragment = (fragment, article, isLead) => {
   // 'enwiki > Quadradic equation' and 'enwiki > Away colors > Association football'). See the
   // 'classifyElements' method itself for other examples.
   requirements.themes.classifyElements(fragment)
+
```

Comment:
After passing through the lazy load transform, what do your placeholders look like? This would be one of the first things to check if the placeholders appeared with the wrong dimensions. If you can copy and paste an image before the transform, after the transform, and after the inverse, I can take a look too. Setting the multiplier to a lower value like .1 should allow you easily examine the different states of a transformed image and verify whether the transform is correct or not. Example here: https://docs.google.com/document/d/1K4bFdzDhSuh2FYs_9XrdD5efW7mMMVvPazWwOctSB_s/edit#heading=h.1ll3i14x7eaw

Reply:
Here's an example from today's featured article. 

Before:
<img width="424" alt="screen shot 2017-12-06 at 12 38 29 pm" src="https://user-images.githubusercontent.com/3143487/33684249-7b35e2d0-da82-11e7-87fc-7b12349e7d63.png">

After:
<img width="424" alt="screen shot 2017-12-06 at 12 38 33 pm" src="https://user-images.githubusercontent.com/3143487/33684250-7b52f6d6-da82-11e7-9902-c4012917c0ac.png">

Reply:
Oh this may be a clue... on `enwiki > Picasso's Blue Period` the first image has the issue, but all of the images after the first one have their placeholders correctly widened... hmm...

Reply:
(In my best Jerry Maguire impression) show me the code! Sorry, I was referring to the raw HTML in each stage (check out the link I shared where you can see how I show what it looks like before the transform, after the transform, and after loading, etc). We can then see if it differs from the demos.

Reply:
lol oops! :) ya i spent a while yesterday live-tweaking the CSS comparing it with your lifecycle examples here https://github.com/wikimedia/apps-android-wikipedia/commit/620538d961221942e340ca7ac7f429393d1309d6

gotta grab some foods atm - will post screenshots after

Reply:
Screenshots from `enwiki > Picasso's Blue Period`

# The first image exhibits the bug:

html w/o lazy loading
![1](https://user-images.githubusercontent.com/3143487/33688535-95a793a8-da90-11e7-835b-78d0a879688a.png)

html w lazy loading, placeholder (shows the bug - i.e. the placeholder dimensions are not widened)
![2](https://user-images.githubusercontent.com/3143487/33688536-95c1d542-da90-11e7-86f1-d226ae4aaf6b.png)

html w lazy loading, image fetched
![3](https://user-images.githubusercontent.com/3143487/33688539-96223004-da90-11e7-85c8-4bdad6b66768.png)


# The second image does not exhibit the bug:

html w/o lazy loading
![4](https://user-images.githubusercontent.com/3143487/33688537-95edf37a-da90-11e7-83dd-9bdcb18a6a58.png)

html w lazy loading, placeholder
![5](https://user-images.githubusercontent.com/3143487/33688538-960bab72-da90-11e7-9393-11460077bb01.png)

html w lazy loading, image fetched
![6](https://user-images.githubusercontent.com/3143487/33688540-96384128-da90-11e7-8975-09fce002c76e.png)

Reply:
Just started looking at this but the first thing I saw was that `pagelib_widen_image_override` appears to be incorrectly limited to `img` in WidenImage.css? If you change the CSS rule from `img.pagelib_widen_image_override` to `pagelib_widen_image_override`, maybe it'll work? (I'm resuming inspection now.)

Reply:
iirc I tried that yesterday and it didn't make any difference. Will try again to make sure...

Reply:
Ok, so changing `img.pagelib_widen_image_override` to `.pagelib_widen_image_override` makes the lazy placeholders disappear completely... wot!

Reply:
^ actually the placeholder's only disappear for images exhibiting the bug - i.e. the 1st image in the screenshots above in this comment thread.

Reply:
I've modified the ThemeTransform demo to perform image widening and lazy loading. I get a significant reflow when I make the screen width very small. Can you try these changes to [WidenImage.css](https://github.com/wikimedia/wikimedia-page-library/pull/110/files#diff-13d24612474e53852200f7cbc0ae483f)?

Reply:
I tried those changes - looks like they break image widening on about half of the images on `enwiki > Pablo Picasso`...

Reply:
^ Looks like the affected images have the `thumbimage` class...

Reply:
I think I have a fix! Still testing, but adding the following to the first line of `widenAncestors` seems to do the trick...
<pre>
if(el.parentElement.tagName === 'A'){
    el.parentElement.style.width = '100%'
}
</pre>